wsd: Rename platform & config specific files

aregnak opened the pull request:
Change-Id: I287310db3c4b7114f8a428a9a9e53afc1acc5dbf


### Summary
Rename the platform and config specific files created in https://github.com/CollaboraOnline/online/pull/12251 to follow the naming convention already present, like the files in `common`.

### Checklist

- [x] I have run `make prettier-write` and formatted the code.
- [x] All commits have Change-Id
- [x] I have run tests with `make check`
- [x] I have issued `make run` and manually verified that everything looks okay

aregnak committed:
wsd: Rename platform & config specific files

Renamed to follow naming convention in common/.

Signed-off-by: Areg Nakashian <areg.nakashian@gmail.com>
Change-Id: I287310db3c4b7114f8a428a9a9e53afc1acc5dbf

kendy commented:
Hi Areg, thanks for the rename, seems it gives a good base for reducing the #ifdefs in the common code!

Two things I have in mind here - would be really good to have a purpose comment for each of the new files, to help others to understand what they are used for; eg. in my case, I was just wondering the other day: Should my changes go to Platform-server.hpp or Platform-unix.hpp? Please nothing too verbose, but something helpful (I'm especially *not* asking for a comment "/* Unix-related includes should go here */" in Platform-unix.hpp :-D ).

Another thing is the SslConfig.hpp. There is only one usage of that file - so I'd just suggest to copy the code out of there into the only user (Platform-server.hpp), and kill that file.

Does that help?

tml1024 commented:
I don't think renaming SslConfig.hpp to Ssl-config.hpp makes sense. The other Foo-bar.hpp files have separate different "bar" versions for each Foo. Like one for server and one for mobile (apps). Having just a single Ssl-config.hpp does not follow the same pattern.  (Plus, what kendy said.)

aregnak committed:
Format Platform files and add purpose comments

I decided to remove Platform-unix since it was only included
in Platform-server, their purpose is much easier to understand
this way for now.

Signed-off-by: Areg Nakashian <areg.nakashian@gmail.com>
Change-Id: I774290eb8f718c01363800fce97ba6f649031210
